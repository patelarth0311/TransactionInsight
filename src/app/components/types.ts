

export type TransactionType = {
    merchant_name: string,
    amount: number,
    date: string,
    category: string[],
    location: {
        region: string
    },
    
}