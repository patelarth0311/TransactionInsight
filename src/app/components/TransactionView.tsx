
import { TransactionType } from "./types"
export function TransactionView({merchant_name,amount,date,category, location}: TransactionType) {
    return <div className="bg-[#121212] flex font-['poppins'] justify-between flex-row w-[500px] shadow h-[80px] rounded-lg p-3 m-5 text-white ">

        <div className="flex flex-col">
        <h1 data-testid="merchant_name">{merchant_name}</h1>
        <h1>{location.region}</h1>
        <h1 data-testid="date">{date}</h1>
        </div>

        <h1 data-testid="amount">   {`$${amount}`}</h1>

        
  

    </div>
}