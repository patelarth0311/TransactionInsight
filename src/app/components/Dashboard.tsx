"use client"
import { SetStateAction, useEffect, useState } from "react"
import { myAxios } from "../axiosConfig"
import { TransactionView } from "./TransactionView"

import * as Realm from "realm-web";
import { useApp } from "../RealmApp"
import { TransactionType } from "./types"

import ChartsEmbedSDK from "@mongodb-js/charts-embed-dom";
import '../panel.css'



export function Dashboard() {
 
    const [accessToken, setAccessToken] = useState('')
    const [transactions,setTransactions]    = useState<TransactionType[]>([])
    const [displayedTransactions,setDisplayedTransactions]    = useState<TransactionType[]>([])

    const sdk = new ChartsEmbedSDK({
        baseUrl: "https://charts.mongodb.com/charts-transactions-bgefp",
      });
      
      const dashboard = sdk.createDashboard( {
        dashboardId: "64d00483-7ddc-4f61-8e90-f6918673d77d", // REPLACE with the Dashboard ID
        height: "1500px",
        widthMode: "scale",
        heightMode: "scale",
        background: 'transparent',
        theme: 'dark'
        // Additional options go here
      })





    useEffect(() => {

        
          
          dashboard.render(document.getElementById("dashboard")!);
     

    },[])





    const app = useApp()


    useEffect(() => {
        if (app?.currentUser) { 
            const mongo = app.currentUser.mongoClient("mongodb-atlas")
            const collection = mongo.db("Portfolio").collection("Transaction"); 
            var fetch = async () => await collection.find({ amount: { $gt: 0  }}).then((res: TransactionType[]) => {
                setTransactions(res)
                setDisplayedTransactions(res.slice(0,5))
    
            })
          
            fetch()
        }
       
    },[app])

    return  <div className="flex items-center  flex-col gap-[10px] p-10 font-['poppins'] ">
        <div className="flex flex-col items-center" >
        <div>
        <h1 className="text-xl m-10 text-left ">Transactions Insights</h1>
        <div id = "dashboard" className="w-screen p-10">

        </div>
        <div id = "chart"></div>

        </div>
  
     
           
        <div className="flex flex-col">
        <h1 className="text-xl m-5 text-left">Latest Transactions</h1>
       
        {displayedTransactions.map((item,index) => (

<TransactionView key={index} merchant_name={item.merchant_name} amount={item.amount} date={item.date} category={item.category} location={item.location } ></TransactionView>

))

} 
        </div>

       
        </div>

    </div>
    
}

