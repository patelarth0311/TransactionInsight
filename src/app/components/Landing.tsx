"use client"

import {
    usePlaidLink,
    PlaidLinkOptions,
    PlaidLinkOnSuccess,
  } from 'react-plaid-link';

import { useEffect } from 'react';
import { useState } from 'react';
import {myAxios} from '../axiosConfig'
import { link } from 'fs';
import { Dashboard } from './Dashboard';
import * as Realm from "realm-web";
import { useApp } from '../RealmApp';

export function Landing() {
    const app = useApp()
    const [user, setUser] = useState()
    const credentials = Realm.Credentials.anonymous();
        // Authenticate the user

        useEffect(() => {
          if (app) {
            login().then((res) => {
              setUser(res)
            })
          }
        },[app])
        var login = async () => await app.logIn(credentials)

    return <div className="font-['poppins'] bg-neutral-50">
     {(app && app.users) &&<Dashboard></Dashboard>
   
     }

    </div>

}

