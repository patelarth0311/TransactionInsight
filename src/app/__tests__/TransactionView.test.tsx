import { render, screen } from '@testing-library/react'
import { describe, it, afterEach } from 'node:test'
import { TransactionView } from '../components/TransactionView'
import { cleanup, } from '@testing-library/react'
import { assert } from 'node:console'
import { create } from 'node:domain'

afterEach(() => {
    cleanup()
})


test("Testing props-passing", () => {

  const transactionView = render(<TransactionView merchant_name={'Uber'} amount={3.41} date={'2023-01-20'} category={["Gaming"]} location={{region: "Georgia"}}/>)

  const merchantH1 = screen.getByTestId("merchant_name");
  const amountH1 = screen.getByTestId("amount");
  const dateH1 = screen.getByTestId("date");

  expect(merchantH1.textContent).toBe("Uber")
  expect(amountH1.textContent?.trim()).toBe("$3.41")
  expect(dateH1.textContent).toBe("2023-01-20")



})

test("Comparing snapshot", () => {  

  const transactionView = render(<TransactionView merchant_name={'Uber'} amount={3.41} date={'2023-09-20'} category={["Gaming"]} location={{region: "Georgia"}}/>)
  expect(transactionView).toMatchSnapshot()
})