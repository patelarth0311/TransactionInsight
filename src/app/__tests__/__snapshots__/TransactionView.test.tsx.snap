// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comparing snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex font-['poppins'] justify-between flex-row w-[500px] shadow h-[80px] rounded-lg p-3 m-5 "
      >
        <div
          class="flex flex-col"
        >
          <h1
            data-testid="merchant_name"
          >
            Uber
          </h1>
          <h1>
            Georgia
          </h1>
          <h1
            data-testid="date"
          >
            2023-09-20
          </h1>
        </div>
        <h1
          data-testid="amount"
        >
             
          $3.41
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex font-['poppins'] justify-between flex-row w-[500px] shadow h-[80px] rounded-lg p-3 m-5 "
    >
      <div
        class="flex flex-col"
      >
        <h1
          data-testid="merchant_name"
        >
          Uber
        </h1>
        <h1>
          Georgia
        </h1>
        <h1
          data-testid="date"
        >
          2023-09-20
        </h1>
      </div>
      <h1
        data-testid="amount"
      >
           
        $3.41
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
